### PR TITLE
♻️ refactor target ratio

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/compounders/slipstream/libraries/SlipstreamLogic.sol [hunk 1]
```diff
@@ -88,4 +88,40 @@ library SlipstreamLogic {
...
+     * @param tickUpper The upper tick of the liquidity position.
+     * @return targetRatio The ratio of the value of token1 compared to the total value of the position, with 18 decimals precision.
+     * @dev Function will revert for all pools where the sqrtPriceX96 is bigger than type(uint128).max.
+     * type(uint128).max is currently more than enough for all supported pools.
+     * If ever the sqrtPriceX96 of a pool exceeds type(uint128).max, a different auto compounder has to be deployed,
+     * which does two consecutive mulDivs.
+     * @dev Derivation of the formula:
+     * 1) The ratio is defined as:
+     *    R = valueToken1 / (valueToken0 + valueToken1)
+     *    If we express all values in token1 en use the current pool price to denominate token0 in token1:
```
Comment (j-vp) [inline src/compounders/slipstream/libraries/SlipstreamLogic.sol:105]:
```suggestion
     *    If we express all values in token1 and use the current pool price to denominate token0 in token1:
```

File: src/compounders/slipstream/libraries/SlipstreamLogic.sol [hunk 2]
```diff
@@ -88,4 +88,40 @@ library SlipstreamLogic {
...
+     */
+    function _getTargetRatio(uint256 sqrtPriceX96, int24 tickLower, int24 tickUpper)
+        internal
+        pure
+        returns (uint256 targetRatio)
+    {
+        uint256 sqrtPriceLower = TickMath.getSqrtRatioAtTick(tickLower);
+        uint256 sqrtPriceUpper = TickMath.getSqrtRatioAtTick(tickUpper);
+
+        uint256 numerator = sqrtPriceX96 - sqrtPriceLower;
```
Comment (j-vp) [inline src/compounders/slipstream/libraries/SlipstreamLogic.sol:122]: will this not underflow if the range is above the current tick?

Comment (Thomas-Smets) [reply src/compounders/slipstream/libraries/SlipstreamLogic.sol:122]: image

**Review by j-vp — DISMISSED** [review]
lgtm

File: src/compounders/slipstream/SlipstreamCompounder.sol [hunk 3]
```diff
@@ -251,17 +251,17 @@ contract SlipstreamCompounder is IActionBase {
             // Swap full amount of token0 to token1.
             zeroToOne = true;
             amountOut = SlipstreamLogic._getAmountOut(position.sqrtPriceX96, true, fees.amount0);
-        } else if (position.currentTick <= position.tickLower) {
+        } else if (position.currentTick < position.tickLower) {
```
Comment (acollette) [inline src/compounders/slipstream/SlipstreamCompounder.sol:254]: That's what you found out recently that price at tick lower is considered in range, but at tick upper considered out of range right ?

**Review by acollette — DISMISSED** [review]
lgtm

File: src/compounders/slipstream/SlipstreamCompounder.sol (hunk 3 above)
Comment (Thomas-Smets) [reply src/compounders/slipstream/SlipstreamCompounder.sol:254]: Yes

**Review by j-vp — APPROVED** [review]

**Review by j-vp — DISMISSED** [review]

**Pull request merged**